File: php/src/Advertisement/Application/Command/PublishAdvertisement/PublishAdvertisementUseCase.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Application\Command\PublishAdvertisement;

use Demo\App\Advertisement\Domain\AdvertisementRepository;
use Demo\App\Advertisement\Domain\Model\Advertisement;
use Demo\App\Advertisement\Domain\Model\ValueObject\Password;

final class PublishAdvertisementUseCase
{
    public function __construct(private AdvertisementRepository $advertisementRepository)
    {
    }

    public function execute(PublishAdvertisementCommand $command): void
    {
        $advertisement = new Advertisement(
            $command->id,
            $command->description,
            Password::fromPlainPassword($command->password),
            new \DateTime(),
        );

        $this->advertisementRepository->save($advertisement);
    }
}

File: kotlin/src/main/kotlin/advertisement/application/publishAdvertisement/PublishAdvertisementUseCase.kt
package advertisement.application.publishAdvertisement

import advertisement.domain.AdvertisementRepository
import advertisement.domain.model.Advertisement
import advertisement.domain.model.value_object.Password
import java.time.LocalDateTime

class PublishAdvertisementUseCase(private val advertisementRepository: AdvertisementRepository) {
    fun execute(addAdvertisementCommand: PublishAdvertisementCommand) {
        val advertisement = Advertisement(
            addAdvertisementCommand.id,
            addAdvertisementCommand.description,
            Password.fromPlainPassword(addAdvertisementCommand.password),
            LocalDateTime.now()
        )

        advertisementRepository.save(advertisement)
    }
}
